src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -385,18 +383,17 @@ public void Unload()
 
         internal static void OnProcessExit()
         {
-            lock (s_contextsToUnload)
+            lock (s_allContexts)
             {
                 s_isProcessExiting = true;
-                foreach (var alcAlive in s_contextsToUnload)
+                foreach (var alcAlive in s_allContexts)
                 {
                     if (alcAlive.Value.TryGetTarget(out AssemblyLoadContext alc))
                     {
-                        // Should we use a try/catch?
-                        alc.InitiateUnload();
+                        alc.RaiseUnloadEvent();
                     }
                 }
-                s_contextsToUnload.Clear();
+                s_allContexts.Clear();
```

Comment:
Do we really need to call `Clear` here?

Reply:
I think we don't have to. The process is going down and so we don't really care. In fact, I was thinking about this stuff over the weekend and I think that we should also remove the `s_isProcessExiting` variable. It just prevents creation of new AssemblyLoadContext after the OnProcessExit is called. In the light of your current PR for removal of special handling of the runtime shutdown, we should not care.

Reply:
I have added a commit removing the s_isProcessExiting and the Clear function call. @jkotas can you please take a look again?